src/test/java/com/pingcap/tikv/dag/TestDAGBuild.java

```diff
@@ -0,0 +1,199 @@
+package com.pingcap.tikv.dag;
+
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.PDClient;
+import com.pingcap.tikv.Snapshot;
+import com.pingcap.tikv.TiConfiguration;
+import com.pingcap.tikv.TiSession;
+import com.pingcap.tikv.catalog.Catalog;
+import com.pingcap.tikv.codec.TableCodec;
+import com.pingcap.tikv.expression.TiByItem;
+import com.pingcap.tikv.expression.TiColumnRef;
+import com.pingcap.tikv.expression.TiConstant;
+import com.pingcap.tikv.expression.aggregate.Count;
+import com.pingcap.tikv.expression.scalar.*;
+import com.pingcap.tikv.kvproto.Coprocessor;
+import com.pingcap.tikv.meta.TiDAGRequest;
+import com.pingcap.tikv.meta.TiDBInfo;
+import com.pingcap.tikv.meta.TiTableInfo;
+import com.pingcap.tikv.region.RegionManager;
+import com.pingcap.tikv.row.Row;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class TestDAGBuild {
+  public static TiConfiguration conf =
```

Comment:
Isn't this test require a real pd deployment?
We need to make sure all unit test feasible in CI environment.
If you need complicated test, either use a mock or move it to integ-test which is not triggered in ci env.

Reply:
Okay, working on it.

Reply:
May move it to integ-test soon.